src/TreeView/TreeView.tsx

```diff
@@ -256,12 +256,28 @@ const Root: React.FC<TreeViewProps> = ({
   flat,
 }) => {
   const containerRef = React.useRef<HTMLUListElement>(null)
+  const mouseDownRef = React.useRef<boolean>(false)
   const [ariaLiveMessage, setAriaLiveMessage] = React.useState('')
   const announceUpdate = React.useCallback((message: string) => {
     setAriaLiveMessage(message)
   }, [])
 
-  useRovingTabIndex({containerRef})
+  const mouseDownHandler = useCallback(() => {
+    mouseDownRef.current = true
+  }, [])
+
+  const mouseUpHandler = useCallback(() => {
+    mouseDownRef.current = false
+  }, [])
+
+  useEffect(() => {
+    document.addEventListener('mouseup', mouseUpHandler)
+    return () => {
+      document.removeEventListener('mouseup', mouseUpHandler)
+    }
+  })
```

Comment:
It seems like the the `mouseUpHandler` is only being used for this `useEffect`. If so, is it possible to rewrite this as a `useEffect` that only runs once by pulling in that handler? e.g.

```suggestion
  useEffect(() => {
    function onMouseUp() {
      mouseDownRef.current = false;
    }
    
    document.addEventListener('mouseup', onMouseUp)
    return () => {
      document.removeEventListener('mouseup', onMouseUp)
    }
  }, [])
```

Reply:
Sure thing! Updated.